core/object/object.h

```diff
@@ -118,8 +118,10 @@ enum PropertyUsageFlags {
 	PROPERTY_USAGE_EDITOR_INSTANTIATE_OBJECT = 1 << 26, // For Object properties, instantiate them when creating in editor.
 	PROPERTY_USAGE_EDITOR_BASIC_SETTING = 1 << 27, //for project or editor settings, show when basic settings are selected.
 	PROPERTY_USAGE_READ_ONLY = 1 << 28, // Mark a property as read-only in the inspector.
+	PROPERTY_USAGE_SECRET = 1 << 29, // Export preset credentials that should be stored separately from the rest of the export config.
```

Comment:
This needs to be bound in `core/core_constants.cpp` after this:
```
core/core_constants.cpp
706:    BIND_CORE_BITFIELD_FLAG(PROPERTY_USAGE_READ_ONLY)
```

Reply:
Done

Reply:
You will also need to add it in the documentation (run doctool, and/or add it manually in `doc/classes/Object.xml` and fill in the description).

Reply:
A question about that. The environment variable strings are global variables currently. If I declared them as `static const String` members in the exporter classes instead, would they show up in the docs? Because I'm still figuring out what the best way to document the available environment variables would be.

Reply:
Sounds good, will do

Reply:
Done. I didn't update the UWP documentation because there is none. I know exporting to UWP doesn't currently work anyway, so I just left it like that.